src/main/java/com/appland/appmap/transform/annotations/CtClassUtil.java

```diff
@@ -33,21 +34,32 @@ public static Boolean isChildOf(CtClass candidateChildClass, String parentClassN
           }
         }
       }
+    }
 
-      CtClass superClass = candidateChildClass.getSuperclass();
-      while (superClass != null) {
-        if (superClass.getName().equals(parentClassName)) {
-          return true;
-        }
-        superClass = superClass.getSuperclass();
+    CtClass superClass = tryClass(candidateChildClass, "superclass", candidateChildClass::getSuperclass);
+    while (superClass != null) {
+      if (superClass.getName().equals(parentClassName)) {
+        return true;
       }
+      final CtClass cls = superClass;
+      superClass = tryClass(cls, "superclass", cls::getSuperclass);
+    }
+
+    return false;
+  }
+
+  private static <V> V tryClass(CtClass cls, String member, ClassAccessor<V> accessor) {
```

Comment:
By testing each access, we don't abort the `isChildOf` entirely if one access attempt raises an exception.